src/proto.c

```diff
@@ -102,8 +102,10 @@ bool get_header(struct MBuf *data, PktHdr *pkt)
 			type = PKT_SSLREQ;
 		} else if (code == PKT_GSSENCREQ) {
 			type = PKT_GSSENCREQ;
-		} else if ((code >> 16) == 3 && (code & 0xFFFF) < 2) {
```

Comment:
to be clear, this was accepting protocol 3.1. It should have been this to only support 3.0:
```diff
		} else if ((code >> 16) == 3 && (code & 0xFFFF) < 2) {
		} else if ((code >> 16) == 3 && (code & 0xFFFF) < 1) {
```